src/shared/fs/watchedFiles.ts

```diff
@@ -275,26 +281,44 @@ export abstract class WatchedFiles<T> implements vscode.Disposable {
     }
 
     /**
-     * Rebuilds registry using current glob and exclusion patterns.
-     * All functionality is currently internal to class, but can be made public if we want a manual "refresh" button
+     * Builds/rebuilds registry using current glob and exclusion patterns. ***Necessary to init registry***.
+     *
+     * @param cancellationPromise Cancels additional registrations when rejected. Otherwise, this
      */
-    public async rebuild(): Promise<void> {
+    public async rebuild(cancellationPromise: Promise<void>): Promise<void> {
```

Comment:
In our codebase we tend to use `Timeout` to represent a cancellationtoken: https://github.com/aws/aws-toolkit-vscode/blob/b061a69854a15f627ff1f0d12a61040179f57eac/src/auth/credentials/utils.ts#L74

its promisify() method may be useful: https://github.com/aws/aws-toolkit-vscode/blob/master/src/shared/utilities/timeoutUtils.ts#L160

Related (but off-topic for this PR): https://taskei.amazon.dev/tasks/IDE-6674